libraries/AP_Scripting/applets/param-set.md

```diff
@@ -0,0 +1,77 @@
+# Parameter Set Filter Script
+
+## Description
+
+This Lua script adds a safty-focused layer to MAVLink parameter setting. When enabled, it intercepts all incoming `PARAM_SET` messages and only allows modification of a predefined whitelist of parameters.
+
+This is especially useful in scenarios where parameter integrity is critical (e.g., automated testing, payload-hosted GCS systems, or shared telemetry environments).
+
+When enabled, the script disables ArduPilot's default GCS handling for parameter writes and takes over that responsibility with additional filtering.
+
+## How It Works
+
+- Intercepts MAVLink `PARAM_SET` messages targeted at the vehicle.
+- Validates that the parameter ID is in the whitelist.
+- Only whitelisted parameters will be updated.
+- Dynamically toggles ArduPilot’s `gcs:set_allow_param_set()` based on enable state.
+
+## Requirements
+
+- ArduPilot 4.7 or later with Lua scripting enabled (`SCR_ENABLE = 1`)
+- The script must be placed in appropriate directories:
+  - Script in: `/scripts/param_set_filter.lua`
```

Comment:
I think this should say /APM/scripts
people will get confused and put this in a scripts directory in root.
it says "scripts in", so should it specify the file name (maybe not)?

Reply:
Changed it